video/cursor.h

```diff
@@ -91,63 +189,100 @@ void cursorDown() {
 	//
 	// Check if scroll required
 	//
-	if (activeCursor->Y + fontH - 1 > activeViewport->Y2) {
-		activeCursor->Y -= fontH;
-		if (~cursorBehaviour & 0x01) {
-			scrollRegion(activeViewport, 3, fontH);
-		}
-		else {
-			activeCursor->X = activeViewport->X2 + 1;
-		}
```

Comment:
this was the previous "scroll protect" implementation.

as can be seen, it doesn't actually really work - yes, it prevents the scroll from happening, and the cursor is left off the right-hand edge of the screen, but the "pending newline" behaviour is absent